### ISSUE TITLE
False RedundantModifier violation for final method of enum inside final class

### ISSUE THREAD

**Issue by fRoSt13iT** [checkstyle/checkstyle#5268]
Here is code:

`TestClass.java`:
```java
public final class TestClass {

    public enum TestEnum {
        VALUE {
           @Override public void someMethodToOverride() {
                // do it differently
            }
         };

        public void someMethodToOverride() { }
        public static final void someStaticMethod() { }
        public final void someMethod() { }         // unexpected violation here
    }

}
```

`config.xml`:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="RedundantModifier"/>
    </module>
</module>
```

Result:
```
java -jar checkstyle-8.4-all.jar -c config.xml TestClass.java 
Starting audit...
[ERROR] /var/tmp/Test.java:11:23: Redundant 'final' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:12:16: Redundant 'final' modifier. [RedundantModifier]
Audit done.
Checkstyle ends with 2 errors.
```

Expected:
only one violation
`[ERROR] /var/tmp/Test.java:11:23: Redundant 'final' modifier. [RedundantModifier]`

https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.9
sub-classing of enum is forbidden but all constants could override methods.
Documentation should have such example as it is not very obvious.

**Comment by romani** [comment]
https://stackoverflow.com/a/14265293

> Since enum can't be subclassed, making a static method final is superfluous but not forbidden.

so please read a round a web about this topic.

> When I remove final modifier from outer class the violation disappears.

Check works as expected.

**Comment by fRoSt13iT** [comment]
The method is not static, so how the answer explain this behavior? I could override this method in enum values, and it is mentioned in documentation of this check.
What is wrong?

**Comment by romani** [comment]
right, 
https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.9

sub-classing of enum is forbidden but all constants could override methods.

This is bug and documentation should be extended to show that nuance.

**Comment by romani** [comment]
@fRoSt13iT , thanks for bringing this back, next time please report issue as it was defined in issue template to avoid misunderstanding on our side.

I updated description to follow rules and explain there is an issue.
All removed all previous comments to keep history of issue clean to be less scary to possible contributor from aside of main team

**Comment by djydewang** [comment]
I am on it.

**Comment by rnveach** [comment]
Fix was merged
